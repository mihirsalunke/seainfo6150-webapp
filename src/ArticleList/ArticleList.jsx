import React from "react";
import ArticleListItem from "../ArticleListItem/ArticleListItem.jsx";

const ArticleList = (props) => {
    return (!props.articles.length) ? <div>You have no data!</div> : (
        <ul>
            {props.articles.map((article) => (<ArticleListItem article={article}></ArticleListItem>))}
        </ul>
      );
};

export default ArticleList;