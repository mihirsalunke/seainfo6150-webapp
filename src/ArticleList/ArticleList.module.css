.articleList {
    list-style: none;
    border: 1px dashed dimgray;
    border-radius: 15px;
    background-color: darkgray;
    box-shadow: 0 0 5px 5px rgba(255, 0, 0, 0.5);
    margin: 15px;
}