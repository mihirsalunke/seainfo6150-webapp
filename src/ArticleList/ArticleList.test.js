import React from "react";
import ArticleList from "./ArticleList";

const data = [
    {
        displayDate: "November 19th 2018, 3:10 am",
        shortText: "I was sitting in an aisle seat on a flight earlier this year when a fellow traveler approached and politely said he thought I may be sitting in his seat.",
        slug: "120103104851-couple-airplane-cabin",
        timeStamp: "2018-11-19T11:10:20.000Z",
        title: "Airline etiquette: Tips from an expert "
    },
    {
        displayDate: "November 22nd 2018, 1:55 am",
        shortText: "From the dramatic Cliffs of Moher to regal Powerscourt Estate in Wicklow, escape to 32 of the most beautiful places to visit in Ireland.",
        slug: "160315093137-beautiful-ireland",
        timeStamp: "2018-11-22T09:55:34.000Z",
        title: "The 32 most beautiful places to visit in Ireland"
    }
];

describe("ArticleList tests", () => {
    it("renders correctly", () => {
        const { container  } = render(<ArticleList articles = {data}/>);
        expect(container ).toMatchSnapshot();
    });
});