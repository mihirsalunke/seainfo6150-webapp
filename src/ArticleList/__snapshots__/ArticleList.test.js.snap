// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <div>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <h2
              class="title"
            >
              Airline etiquette: Tips from an expert 
            </h2>
          </div>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <h2
              class="title"
            >
              The 32 most beautiful places to visit in Ireland
            </h2>
          </div>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
    </ul>
  </div>
</div>
`;
